### ISSUE TITLE
Implement Algorithmic Constituency

### ISSUE THREAD

**Issue by domi41** [MieuxVoter/mvapi#29]
Hello !

Would you be open to a test-suite written in plain english (and/or french) that would run integration tests on the API?

It would probably leverage [behave](https://behave.readthedocs.io/en/latest/).

**These features act like an algorithmijc constitution**, sort of.  It allows informatically illiterate people to read and contribute to the specifications of the application, as well as reporting bugs in a very convenient fashion.  _It shifts the sovereignty of the application from the developers to the community._

This is a pretty big and humiliating decision for developers to take, so we'd understand if you don't want to.

The features would look like this  (working draft, in no way finished, and after re-reading it, it lacks some serious, but perhaps you'll get the idea) : https://framagit.org/limaju/limaju-server-features-fr/-/blob/master/10.fonctionnalites/20.voter-sur-un-scrutin.feature  (we would rewrite them entirely for this project, and with consensus, of course)

---

From previous projects, we noted that the idiomatic features are best put in their own repository (one per language), and the feature runner, context and step defs in the project repository itself (here).

This allows Issues and Wiki and other repository goodies to be scoped correctly, and it also allows reusing the features on multiple projects, for example when this API will be rewritten in Go or Elixir or Scala to improve performance.

If you're interested in this, and you'd like to peer-code it, we can set this all up together in the near future, using one of the many screen-sharing, VoIP software available.

**Comment by guhur** [comment]
That sounds really nice :) 
Not course I grasped how it works exactly. So maybe we can schedule a meeting first to talk about it?

Did you join Mieux Voter Telegram channel? Otherwise, what is your favorite communication tool?

**Comment by domi41** [comment]
> That sounds really nice :)

Glad you think so !

> Did you join Mieux Voter Telegram channel?

Telegram isn't working on our computer, for some reason.

> Otherwise, what is your favorite communication tool?

The Fediverse (`@DominiqueMerle@mastodon.social`), Email (dominique.merle.41@gmail.com), Discord (…sigh), Riot (<3), Mumble (<3 but no screenshare), Retroshare, shared `screen` in CLI, and sometimes we squat one of the instances of Jitsi Meet out there.  We can also boot some IRC client back up.

Contact info may be redacted :zipper_mouth_face: from this message once we've connected. 
This does not feel overly safe.
